mapie/time_series_regression.py

```diff
@@ -0,0 +1,325 @@
+from ._typing import ArrayLike, NDArray
+from .utils import (
+    check_alpha,
+    check_alpha_and_n_samples,
+    masked_quantile,
+)
+
+
+class MapieTimeSeriesRegressor(MapieRegressor):
+    """
+    Prediction interval with out-of-fold residuals for time series.
+
+    This class implements the EnbPI strategy and some variations
+    for estimating prediction intervals on single-output time series.
+
+    Actually, EnbPI only corresponds to ``MapieTimeSeriesRegressor`` if the
+    ``cv`` argument if of type ``BlockBootstrap``.
+
+    References
+    ----------
+    Chen Xu, and Yao Xie.
+    [6] "Conformal prediction for dynamic time-series."
+    https://arxiv.org/abs/2010.09107
+    """
+
+    def __init__(
+        self,
+        estimator: Optional[RegressorMixin] = None,
+        method: str = "plus",
+        cv: Optional[Union[int, str, BaseCrossValidator]] = None,
+        n_jobs: Optional[int] = None,
+        agg_function: Optional[str] = "mean",
+        verbose: int = 0,
+    ) -> None:
+        super().__init__(estimator, method, cv, n_jobs, agg_function, verbose)
+        self.cv_need_agg_function.append("BlockBootstrap")
+
+    def fit(
+        self,
+        X: ArrayLike,
+        y: ArrayLike,
+        sample_weight: Optional[ArrayLike] = None,
+    ) -> MapieTimeSeriesRegressor:
+        """
+        Returns
+        -------
+        MapieTimeSeriesRegressor
+            The model itself.
+        """
+        self = super().fit(X=X, y=y, sample_weight=sample_weight)
+        y_pred, _ = super().predict(X, alpha=0.5, ensemble=True)
+        self.conformity_scores_ = np.asarray(y) - y_pred
+        return self
+
+    def partial_fit(
+        self,
+        X: ArrayLike,
+        y: ArrayLike,
+    ) -> MapieTimeSeriesRegressor:
+        """
+        Update the ``conformity_scores_`` attribute when data with known labels
+        are available.
+
+        Parameters
+        ----------
+        X : ArrayLike of shape (n_samples, n_features)
+            Input data.
+
+        y : ArrayLike of shape (n_samples,)
+            Input labels.
+
+        Returns
+        -------
+        MapieTimeSeriesRegressor
+            The model itself.
+        """
+        y_pred, _ = self.predict(X, alpha=0.5, ensemble=True)
+        new_conformity_scores_ = np.asarray(y) - np.asarray(y_pred)
+        new_conformity_scores_ = new_conformity_scores_[
+            ~np.isnan(new_conformity_scores_)
+        ]
+
+        cut_index = min(
+            len(new_conformity_scores_), len(self.conformity_scores_)
+        )
+        self.conformity_scores_ = np.concatenate(
+            [
+                self.conformity_scores_[cut_index:],
+                new_conformity_scores_,
+            ],
+            axis=0,
+        )
+        return self
+
+    def predict(
+        self,
+        X: ArrayLike,
+        ensemble: bool = False,
+        alpha: Optional[Union[float, Iterable[float]]] = None,
+        JAB_Like=False,
```

Comment:
Not very pep8-friendly, I would change it to "jab_like".

Reply:
Add the type

Reply:
Change